src/QuartoGameTransitionGenerator.java

```diff
@@ -0,0 +1,76 @@
+import java.util.Iterator;
+import java.util.HashMap;
+import java.util.ArrayList;
+
+/**
+ * Created by brendend on 15-03-14.
+ */
+public class QuartoGameTransitionGenerator implements Iterable<QuartoGameTransition>{
+
+    public QuartoGameState fromState;
+    public QuartoPiece limboPiece;
+    public Iterator<QuartoPiece> pieces;
+
+
+    QuartoGameTransitionGenerator(QuartoGameState fromState, QuartoPiece limboPiece) {
+        this.fromState = fromState;
+        this.limboPiece = limboPiece;
+        ArrayList<QuartoPiece> tempPieces = (ArrayList<QuartoPiece>)fromState.freePieces.clone();
+        tempPieces.remove(limboPiece);
+        pieces = tempPieces.iterator();
+
+    }
+
+
+    /**
+     * @return       Iterator<QuartoGameState>
+     */
+    @Override
+    public Iterator<QuartoGameTransition> iterator()
+    {
+        return new Iterator<QuartoGameTransition>() {
+            private QuartoGameState curState = fromState;
+            private Iterator<QuartoGameTransition> transitions = curState.transitions.values().iterator();
+            private Iterator<int[]> squares = curState.freeSquares.iterator();
+            private QuartoPiece nextPiece = null;
+            private int[] nextSquare = null;
+
+            @Override
+            public QuartoGameTransition next() {
+                if(transitions.hasNext()) {
```

Comment:
Just making note here so we don't forget but we're likely going to duplicate our efforts by looping through transitions, then by creating all of the possible children (i.e. if we explore a transition, we'll then re-create it and attempt to explore it again)

This will be "fixed" by Randy's work on not exploring a node multiple times from a given root, but we should also try to not iterate over it again.